desk/lib/logs.hoon

```diff
@@ -0,0 +1,57 @@
+/-  *logs
+|%
+::
+++  fail-event
+  |=  [=term =tang]
+  ^-  $>(%fail log-event)
+  [%fail term tang]
+::
+++  log-fail
+  |=  [=wire our=ship event=$>(%fail log-event)]
+  ^-  card:agent:gall
+  [%pass wire %agent [our %logs] %poke log-action+!>([%log event])]
+::
+++  enjs
+  =,  format
+  |%
+  ++  tang
+    |=  t=^tang
+    ^-  $>(%a json)
+    ?~  t  a+~
+    =/  tame=(list tape)
+      %+  turn  t
+     (cork (cury wash [0 80]) zing)
+    ::XX posthog does not display newlines properly
+    :: s+(crip (zing `(list tape)`(join "\0a" tame)))
+    a+(turn tame tape:enjs)
```

Comment:
You want to preserve the newlines here, so I don't think you want to `+zing`. Or rather, you probably want to do that _later_:

```suggestion
    =/  tame=(list tape)
      %-  zing
      %+  turn  t
      (cury wash [0 80])
    ::XX posthog does not display newlines properly
    :: s+(crip (zing `(list tape)`(join "\0a" tame)))
    a+(turn tame tape:enjs)
```